packages/dds/merge-tree/src/localReference.ts

```diff
@@ -132,6 +132,10 @@ export class LocalReferenceCollection {
         if (seg2.localRefs && !seg2.localRefs.empty) {
             if (!seg1.localRefs) {
                 seg1.localRefs = new LocalReferenceCollection(seg1);
+            } else if (seg1.cachedLength > seg1.localRefs.refsByOffset.length) {
```

Comment:
the goal was to always have refsByOffset.length === cachedLength. I see how we are diverging, but i'm a bit worried about late binding the fixup here, as there are a lot of places, we enumerate refsByOffset.length.

Reply:
We could instead bump the length every time we append a segment with no references to a segment that already has a non-empty LocalReferenceCollection.

Reply:
```
    public static append(seg1: ISegment, seg2: ISegment) {
        if (seg2.localRefs && !seg2.localRefs.empty) {
            if (!seg1.localRefs) {
                seg1.localRefs = new LocalReferenceCollection(seg1);
            } 
            seg1.localRefs.append(seg2.localRefs);
        }else{
                if(seg1.localRefe){
                        seg1.refsByOffset.length += seg2.cachedLength
                }
        }
    }
```

Reply:
i would even be ok adding an assert in the condition before we increment that `seg1.refsByOffset.length === seg1.cachedLength`